clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1288,6 +1288,9 @@ else if (req.isInternalRequest && body instanceof ApiVersionsResponse)
                 receive.close();
             }
         }
+
+        // Clear the completed receives once they have been added to ClientResponse returned via poll()
+        this.selector.completedReceives().clear();
```

Comment:
On each poll, the first thing it does is to clear everything https://github.com/linkedin/kafka/blob/2.4-li/clients/src/main/java/org/apache/kafka/common/network/Selector.java#L760.  Why do we need to clear it twice?

Reply:
Same reason as the original PR #306 
>This is to avoid returning the same ByteBuffer to MemoryPool twice. If the close is called but the ByteBuffer will be returned twice to MemoryPool by close in NetworkClient and the close by CompletedReceive in Fetcher

https://github.com/linkedin/kafka/pull/306#discussion_r828562709